store/commitment/store.go

```diff
@@ -0,0 +1,151 @@
+package commitment
+
+import (
+	"errors"
+	"fmt"
+
+	dbm "github.com/cosmos/cosmos-db"
+	ics23 "github.com/cosmos/ics23/go"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/commitment/iavl"
+	"cosmossdk.io/store/v2/commitment/types"
+)
+
+const (
+	// DefaultTreeType is the default tree type.
+	DefaultTreeType = "default"
+	IavlTreeType    = "iavl"
+)
+
+var _ store.Committer = (*CommitStore)(nil)
+
+// CommitStore is a wrapper around the state Tree.
```

Comment:
nit: This is the core abstraction and it's very important IMO. It deserves a pretty detailed godoc.

```suggestion
// CommitStore is a wrapper around multiple Tree objects mapped by a unique store
// key. Each store key reflects dedicated and unique usage within a module. A caller
// can construct a CommitStore with one or more store keys. It is expected that a
// RootStore use a CommitStore as an abstraction to handle multiple store keys
// and trees.
```